Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -205,13 +210,40 @@ void APSStarActor::SetStarActorProgressMeshMaterial(class UMaterialInstanceDynam
 		return; // Early return if pointers are invalid
 	}
 
-	// Add a face texture over current star material
-	const FPSSettingsRow& CurrentSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRow();
-	const FName StarFaceTextureParameter = UPSDataAsset::Get().GetStarFaceTextureParameter();
+	const UPSSpotComponent* SpotComponent = UPSWorldSubsystem::Get().GetCurrentSpot();
+	if (!ensureMsgf(SpotComponent, TEXT("ASSERT: [%i] %hs:\n'SpotComponent' is not valid!"), __LINE__, __FUNCTION__))
+	{
+		return; // Early return if pointers are invalid
+	}
+
+	const FPlayerTag CurrentSpotPlayerTag = SpotComponent->GetMeshChecked().GetPlayerTag();
 	const FName StarProgressionMaterialSlotName = UPSDataAsset::Get().GetStarMaterialSlotName();
-	UTexture* StarFaceTexture = CurrentSettingsRowData.StarFaceTexture;
 
-	StarMeshComponent->SetMaterial(0, StarDynamicMaterial);
-	StarDynamicMaterial->SetScalarParameterValue(StarProgressionMaterialSlotName, StarProgressionAmount);
-	StarDynamicMaterial->SetTextureParameterValue(StarFaceTextureParameter, StarFaceTexture);
+	ChangeStarMesh(SpotComponent);
+
+	UMaterialInstanceDynamic* DynamicOverlay = UPSWorldSubsystem::Get().GetOverlayProgressionDynamicMaterialByTag(CurrentSpotPlayerTag);
+	DynamicOverlay->SetScalarParameterValue(StarProgressionMaterialSlotName, StarProgressionAmount);
+	StarMeshComponent->SetOverlayMaterial(DynamicOverlay);
+}
+
+// Changes current bomb mesh to current spot bomb mesh
+void APSStarActor::ChangeStarMesh(const UPSSpotComponent* SpotComponent)
+{
+	if (!ensureMsgf(SpotComponent, TEXT("ASSERT: [%i] %hs:\n'SpotComponent' is not valid!"), __LINE__, __FUNCTION__))
+	{
+		return; // Early return if pointers are invalid
+	}
+
+	const ELevelType SpotType = SpotComponent->GetMeshChecked().GetAssociatedLevelType();
+	const ULevelActorRow* BombRow = UBombDataAsset::Get().GetRowByLevelType(SpotType);
+	UStaticMesh* BombMesh = BombRow ? Cast<UStaticMesh>(BombRow->Mesh) : nullptr;
+	if (!ensureMsgf(BombMesh, TEXT("ASSERT: [%i] %hs:\n'BombMesh' is not valid!"), __LINE__, __FUNCTION__))
+	{
+		return; // Early return if pointers are invalid
+	}
+
+	UMaterialInterface* Material = BombMesh->GetMaterial(0);
+
+	StarMeshComponent->SetStaticMesh(BombMesh);
+	StarMeshComponent->SetMaterial(0, Material);
```

Comment:
Explicitly assigning material from the mesh that you just assigned doesn't make much sense: this is unusual since the mesh's default material should apply automatically. It indicates that the component likely has a leftover material override. Instead, reset the material first by assigning `nullptr`:

Before:
```
	UMaterialInterface* Material = BombMesh->GetMaterial(0);

	StarMeshComponent->SetStaticMesh(BombMesh);
	StarMeshComponent->SetMaterial(0, Material);
```
After:
```
	StarMeshComponent->SetMaterial(0, nullptr);
	StarMeshComponent->SetStaticMesh(BombMesh);
```

Reply:
done as part of https://github.com/moinm3uw/ProgressionSystem/pull/66/commits/664c6dfd52822e2eee472e4bb41afea40d365b0e